src/VolumeCache.ts

```diff
@@ -0,0 +1,306 @@
+type XYZ<T> = { x: T; y: T; z: T };
+export type VolumeScaleDims = XYZ<number>;
+type CacheEntryExtent = XYZ<[number, number]>;
+
+// The following two very similar types are kept separate because we may later want to allow more
+// complex queries with respect to scale, e.g. "get the largest available scale within this range"
+export type DataArrayExtent = CacheEntryExtent & {
+  scale: number;
+  time: number;
+  channel: number;
+};
+
+export type QueryExtent = CacheEntryExtent & {
+  scale: number;
+  time: number;
+};
+
+type CacheEntry = {
+  /** The data contained in this entry */
+  // TODO allow more types of `TypedArray` to be stored together in the cache?
+  data: Uint8Array;
+  /** The subset of the volume covered by this entry */
+  size: CacheEntryExtent;
```

Comment:
consider renaming to something like `bounds` or `extent` or even `size3d` because `size` makes me think of scalar number of bytes.